### PR TITLE
Full Auto Fixes

### PR DESCRIPTION
**Pull request by Geevies**
* Fixes a lot of weirdness surrounding guns on full-auto mode. Safety checks will now be skipped on harm intent, the gun will stop shooting when dropped, and putting the gun in a storage item will no longer shoot it.

### CODE REVIEW

**Review by NonQueueingMatt — APPROVED** [review]
this is a bit concerning for lag so I guess we'll see how it goes

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -836,15 +836,61 @@ var/list/localhost_addresses = list(
...
+		var/accuracy_dec = 0
+		while(autofire_aiming_at[1])
+			// stop shooting if we drop our gun
+			if(G.loc != mob)
+				break
+			G.Fire(autofire_aiming_at[1], mob, autofire_aiming_at[2], (get_dist(mob, location) <= 1), FALSE, accuracy_dec)
+			mob.set_dir(get_dir(mob, autofire_aiming_at[1]))
+			accuracy_dec = min(accuracy_dec + 0.25, 2)
+			sleep(G.fire_delay)
 			CHECK_TICK
```
Comment (FluffyGhoster) [inline code/modules/client/client_procs.dm:894]: Yelding to CHECK_TICK from a proc that's meant to handle user inputs seems like a terrible idea, I understand however that you didn't wrote this, but sinde there's a sleep, this CHECK_TICK seems between bad and pointless, even sleep(0) would yeld until the end of the tick?

Comment (Geevies) [reply code/modules/client/client_procs.dm:894]: i'm not sure tbh, i'm not too in the loop with how ticks and stuff works, so if you have a recommendation, lemme know

Comment (FluffyGhoster) [reply code/modules/client/client_procs.dm:894]: I'm 98% sure it shouldn't be there and can / is better to remove it; comparing with other codebases, neither TG nor Bay have it either

Comment (FluffyGhoster) [inline code/modules/client/client_procs.dm:894]:
```suggestion
```

image

**Review by FluffyGhoster — APPROVED** [review]
Looks good to go in, nothing exploded

**Pull request merged**